dagql/idtui/frontend.go

```diff
@@ -147,16 +147,18 @@ func (r renderer) renderCall(out *termenv.Output, span *Span, call *callpbv1.Cal
 				val := arg.GetValue()
 				fmt.Fprint(out, " ")
 				if argDig := val.GetCallDigest(); argDig != "" {
-					var argSpan *Span
-					var argInternal bool
-					if span != nil {
-						argSpan = r.db.MostInterestingSpan(argDig)
-						if argSpan != nil {
-							argInternal = argSpan.Internal
+					forceSimplify := false
+					internal := internal
+					argSpan := r.db.MostInterestingSpan(argDig)
+					if argSpan != nil {
+						forceSimplify = argSpan.Internal && !internal // only for the first internal call (not it's children)
```

Comment:
Checking I have this right: if we're rendering a non-internal call, force simplification for internal args. But if we're already rendering an internal call, there's no need to simplify internal args, since we're already internal.

Reply:
Yes :laughing: sorry, I'm kinda struggling if there's an easier way to write this :cry: